lib/packwerk.rb

```diff
@@ -13,44 +13,74 @@
 module Packwerk
   extend ActiveSupport::Autoload
 
+  # Public APIs
```

Comment:
I chose these APIs with these rationales:

### Checker
API for extending checkers

### Cli
Main API for calling packwerk right now. Eventually we might want API like `Packwerk.check` instead of using the CLI, but this is the de-facto API.

### Offense
Part of the interface for `OffensesFormatter`

### OffensesFormatter
API for extending offenses formatter

### OutputStyle
Implementation of this needs to be passed into CLI

### Package, PackageSet, PackageTodo
API for getting information about YML files

### Parsers
Contains documented API for `Packwerk::Parsers::Factory.erb_parser_class=(...)`, which is the current API for substituting the ERB parser.
The hope is that https://github.com/Shopify/packwerk/pull/243 will allow us to mark this API as private.

This module also contains `ParserInterface` as well. With the above PR, we'd allow the interface to be public but the rest to be private.

### Reference, ReferenceOffense
Part of the API to `Checker`

### Result
Return value of CLI

Reply:
Result is overloaded, I vote to nest the toplevel result in CLI. We should also nest the app validator result in Validator instead, especially if we use it in the interface.